appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/XPathToDeveloperProvidedContentRefs.java

```diff
@@ -221,12 +221,14 @@ void addToContentIfInApp(
                         new Object[] {referringDocument, pathToContent});
             } else {
                 final ApplicationSignedJARManager signedJARManager = helper.signedJARManager();
-                if (signedJARManager == null && helper instanceof NestedAppClientDeployerHelper) {
```

Comment:
Problem is the `&&` that makes null be possible in other branches but its not expected to be. I believe what was intended here is to only log an error for null `signedJARManager` in case of a `NestedAppClientDeployerHelper`. Should any other helper occur with null this goes unnoticed but from the comments it sounds like we expect this to be a nested client helper. We might want to do some other logging or we just be satisfied as it is. What previously was ending with a NPE now has no log entry which seems reasonable since nobody had reported a NPE for this location so I would guess its not possible or common.